### PR TITLE
been changed: title of the project, add read edit and delete of CRUD.

### PR DESCRIPTION
**Pull request by roman89ua**
changes: - project title; -subject read; -subject delete; -subject edit;

### CODE REVIEW

File: src/app/admin/subjects/subject-confirm/subject-confirm.component.html [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<h1 mat-dialog-title>Підтвердження</h1>
+<div mat-dialog-content>
+    <p>{{confirm}}</p>
+</div>
+<div mat-dialog-actions>
+    <button mat-button color="primary" (click)="onCensle()">Скасувати</button>
```
Comment (tatkaif) [inline src/app/admin/subjects/subject-confirm/subject-confirm.component.html]: Looks like there is a mistake here `onCensle()`

File: src/index.html [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
 <head>
-  <meta charset="utf-8">
-  <title>IF105</title>
-  <base href="/">
-  <meta name="viewport" content="width=device-width, initial-scale=1">
-  <link rel="icon" type="image/x-icon" href="favicon.ico">
-  <link href="https://fonts.googleapis.com/css?family=Roboto:300,400,500&display=swap" rel="stylesheet">
-  <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
+    <meta charset="utf-8">
+    <title>deTester</title>
```
Comment (tatkaif) [inline src/index.html]: d-tester

**Review by roman89ua — COMMENTED** [review]
Done with mistakes

File: src/index.html (hunk 2 above)
Comment (roman89ua) [reply src/index.html]: Corrected

File: src/app/admin/subjects/subject-confirm/subject-confirm.component.html (hunk 1 above)
Comment (roman89ua) [reply src/app/admin/subjects/subject-confirm/subject-confirm.component.html]: Corrected

File: src/app/admin/subjects/subject-confirm/subject-confirm.component.html [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<h1 mat-dialog-title>Підтвердження</h1>
+<div mat-dialog-content>
+    <p>{{confirm}}</p>
+</div>
+<div mat-dialog-actions>
+    <button mat-button color="primary" (click)="onCansle()">Скасувати</button>
```
Comment (AndriyIf) [inline src/app/admin/subjects/subject-confirm/subject-confirm.component.html]: Still not correct. `onCancel`

**Review by AndriyIf — APPROVED** [review]

File: src/app/admin/subjects/subject-confirm/subject-confirm.component.html (hunk 3 above)
Comment (roman89ua) [reply src/app/admin/subjects/subject-confirm/subject-confirm.component.html]: Done:)

**Review by tatkaif — APPROVED** [review]

File: src/app/admin/subjects/subjects-create-modal/subjects-create-modal.component.html [hunk 4]
```diff
@@ -1,6 +1,7 @@
-<form class="input-wrapper" [formGroup]="addSubject">
+<form class="input-wrapper" [formGroup]="addSubject" autocomplete="off">
     <mat-form-field class="example-full-width">
```
Comment (LiubomyrTs) [inline src/app/admin/subjects/subjects-create-modal/subjects-create-modal.component.html:2]: I suggest using appropriate css classes (not 'example-full-width' but 'full-width')

File: src/app/admin/subjects/subjects.component.ts [hunk 5]
```diff
@@ -39,10 +55,52 @@ export class SubjectsComponent implements OnInit {
           return of(null);
         })
       )
-      .subscribe((newData: SSubjects | null) => {
+      .subscribe((newData: Array<ISubjects> | null) => {
+        if (newData) {
+          this.subjectTableList = [...this.subjectTableList, ...newData];
```
Comment (LiubomyrTs) [inline src/app/admin/subjects/subjects.component.ts:60]: seems like this.subjectTableList.push(newData) can be used

File: src/app/admin/subjects/subjects.component.ts [hunk 6]
```diff
@@ -39,10 +55,52 @@ export class SubjectsComponent implements OnInit {
...
+
+  delete(row: ISubjects): void {
+    const dialogData = `Ви видаляєте ${row.subject_name}, ви впевнені?`;
+    const dialogRef = this.dialog.open(SubjectConfirmComponent, {
+      maxWidth: '400px',
+      data: dialogData
+    });
+    dialogRef.afterClosed().subscribe(dialogResult => {
+      this.result = dialogResult;
+      if (this.result) {
```
Comment (LiubomyrTs) [inline src/app/admin/subjects/subjects.component.ts:95]: why are you storing this.result?, why not simply use if (dialogResult) {}

Comment (LiubomyrTs) [reply src/app/admin/subjects/subjects.component.ts:95]: you can do it in a fancier way 
`dialogRef.afterClosed(
 filter(res => res), 
 switchMap(() => this.subjectService.deleteSubject(row.subject_id))).subscribe((res) => this.subjectTableList = this.subjectTableList.filter(item => item.subject_id !== row.subject_id);)`

but it's up to you

**Review by LiubomyrTs — APPROVED** [review]

File: src/app/admin/subjects/subjects.component.ts (hunk 5 above)
Comment (roman89ua) [reply src/app/admin/subjects/subjects.component.ts:60]: Alredy fixed this in a new wersion. I use just a showSubjects function to show new resuts

**Pull request merged**